src/Common/src/TypeSystem/Common/MetadataFieldLayoutAlgorithm.cs

```diff
@@ -407,11 +407,17 @@ protected static ComputedInstanceFieldLayout ComputeSequentialFieldLayout(Metada
             return computedLayout;
         }
 
-        protected static ComputedInstanceFieldLayout ComputeAutoFieldLayout(MetadataType type, int numInstanceFields)
+        protected virtual void AlignBaseOffsetIfNecessary(MetadataType type, ref LayoutInt baseOffset)
+        {
+        }
+
+        protected ComputedInstanceFieldLayout ComputeAutoFieldLayout(MetadataType type, int numInstanceFields)
         {
             // For types inheriting from another type, field offsets continue on from where they left off
             LayoutInt cumulativeInstanceFieldPos = ComputeBytesUsedInParentType(type);
 
+            AlignBaseOffsetIfNecessary(type, ref cumulativeInstanceFieldPos);
```

Comment:
Do we only need this for Auto field layout? Not for sequential?

Reply:
I may be mistaken but after rereading the CoreCLR MethodTableBuilder code one more time I'm convinced it only applies to auto layout. NeedsAlignedBaseOffset is only called from PlaceInstanceFields which I believe to only handle auto layout; sequential and explicit layout are handled in HandleExplicitLayout which doesn't include any such check.